### ISSUE TITLE
[Design] Connected accounts updates

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#6731]
## User Story

As a user who has allowed a 3rd-party application to access my VA.gov account, I need to be able to access and manage these permissions quickly and easily from VA.gov

## Additional background

Carey works with the VA API team and will be managing connected accounts going forward. She would like to make some updates for profile 2.0. We'll use this ticket to track the proposed design updates.

[Carey's Figma prototype](https://www.figma.com/proto/Za0usVKp8l80mlrsx7FaB0/Account?node-id=13%3A1690&scaling=min-zoom)

## Tasks

- [x] Carey to share design patterns/interactions you'd like to see @tressaellen implement
- [x] Iterate design with team
- [x] Settle on a final proposal

**We can create a separate ticket to get copy help from Peggy or someone else on the team*.

## Acceptance Criteria

- [x] Carey, Tressa, and Samara are in agreement about the connected accounts direction

**Comment by Samara-Strauss** [comment]
@tressaellen @mshea0606 we'll be using this ticket to track Carey's design updates. @-ing you here so you're subscribed to the convo.

**Comment by Samara-Strauss** [comment]
Moving conversations over from Slack: 

From me:

**On the designs themselves**

1. Could we document this conversation in Github somewhere so it's easier to track your proposed updates and any comments/discussions that result from it?
2. The empty state is confusing to me — if I've never seen this and I see something that says "no one has access to your data", I'd be worried.
3. Why does a user need to see previously connected apps?
4. Generally speaking, the other interactions all look good/make sense to me.
5. Your copy changes between some of the screens where I don't think you intend it to change. Might want to fix that so it's easier for others to give you feedback.
6. Next steps: Can you get this into a github ticket? Once the design is finalized, can you create a ticket for copy needs and work with one of the content folks? And please include @Tressa Furner on the ticket so she can follow along.

**What we might include in Profile 2.0 launch vs. follow-up work**

1. All copy updates should be easy peasy.
2. If you want to include FAQs, can you please lead the work to get these ready and on staging?
3. For the empty state, let's get this into a good place in the design before we determine whether we want to include it in the MVP. If it's simple enough and provides users information on how to add accounts, this should be fine.
4. I don't think we have a list of previously connected accounts right now. If this is new functionality, it would need to be follow-up work after Profile 2.0 launches.

And Carey's response:

1. Our team is now using Jira for tickets, not GitHub. Is there a particular place in GitHub you’d like me to post this for cross-team collaboration?
2. Re: copy on empty state - I took tone from what I was seeing with Google, but please consider exact wording placeholder for now. Would LOVE help with this from your copywriting team and see this implemented with the Profile 2.0 work. We will likely want to strike a balance between encouraging users to trust the technology and encouraging them to be defensive/proactive with their privacy. Besides better copy and FAQ page, what else do you need to see in order to get this implemented into the MVP?
3. Showing a user previously connected apps is similar to audit logging, a security-relevant chronological record of a users’ data. We have anecdotal evidence from our Medicare counterparts that users want this, because in tech things are often impermanent or changing. Also, disconnecting an app doesn’t always mean your data with that app is purged, so a record of what you’ve been connected to could be helpful when a user wants to clean house. I agree this would fall in line with new functionality/follow-up work which our team will lead. We can do user testing around it to validate the usefulness before implementing.
4. Re: changing copy between screens - was playing around with different ideas for tone. Again, would love Peggy’s feedback on this.
5. Re: FAQs, yes, will lead this work.

**Comment by Samara-Strauss** [comment]
@careyotto here are my thoughts on your latest response from Slack (which I added in the comment above ☝️ )


1. Looks like we got the Github issue taken care of!
2. Re: copy on empty state - Yes, please work with Peggy or someone else on the content team (we can loop them in when we are ready). For the empty state, I think you need to (a) explain what the feature is, (b) link to an FAQ for more info, (c) show no apps are connected. Will definitely consider exact copy placeholder for now, but it would be helpful if what you included in your designs was in the spirit of what the copywriter should write. We can adjust that placeholder copy until we feel like it communicates the right idea, and then an actual content person can finesse it to make sure it fits into our style guidelines.
3. _Besides better copy and FAQ page, what else do you need to see in order to get this implemented into the MVP?_ We need the design patterns/interactions you'd like to see @tressaellen implement (I think we get a good idea of that above), exact copy, and the FAQ page (if the FAQ page will be MVP). If the FAQ page will not be in the MVP, the designs we're working with now should reflect that.
4. _Showing a user previously connected apps is similar to audit logging, a security-relevant chronological record of a users’ data...I agree this would fall in line with new functionality/follow-up work which our team will lead. We can do user testing around it to validate the usefulness before implementing._ Since this is not existing functionality, let's take it out of the mockups we're doing for Profile 2.0. This should be something you/your team works on/validates through user testing, and then we can talk about getting it into the profile.
5. _Re: changing copy between screens - was playing around with different ideas for tone. Again, would love Peggy’s feedback on this._ Totally understand. Still, having one set of copy will be much easier and less confusing. Otherwise, Peggy/anyone else may get confused and think they have to write extra copy they don't have to write if they see copy changes between the second and third screens of the prototype. When you're working with Peggy/someone else, all they need is an idea of what your copy needs are and the general sentiments that need to be communicated.
6. _Re: FAQs, yes, will lead this work._ Great! Can you please track this on Github as well in a different ticket in this repo so we can follow along more easily? You'll need to work with the public websites team to get this done.

**Comment by mshea0606** [comment]
Summary of what we need from Carey:

- Design patterns/interactions you'd like to see @tressaellen implement 
- Finalized copy from Peggy 
- The FAQ page (if the FAQ page will be MVP).  If the FAQ page will NOT be in the MVP, the designs 
   we're working with now should reflect that. 
- Ticket in Github to create FAQ page (Public Websites team will help create this)
- Eliminate showing users "previously connected apps" for Profile 2.0 (can be added later)

**Comment by Samara-Strauss** [comment]
@mshea0606 thanks for this! can you please update the first comment in this ticket with any of those to-dos that are part of this ticket for easy tracking? I don't want to-do items to get lost buried in the discussion.

Also, we should have separate tickets for the FAQ page and the copy for connected accounts. @mshea0606 @careyotto can one of you please create those tickets? Those do not need to be t-do items that are part of the requirements to close this ticket out.

**Comment by careyotto** [comment]
Updates:
- Copy and screens in the prototype have been updated, based on everyone's feedback. 
- [created a ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6748) for @peggygannon to review UI copy on all the screens. 
- @tressaellen Will be taking this prototype and incorporating the ideas behind it into her work.
- I have a call with Aruna Kucherlapati tomorrow to discuss getting FAQ type page up on VA.gov. Will created relevant tickets related to that work after discussing with Aruna.

**Comment by careyotto** [comment]
@mshea0606 I've updated the tasks at the top. Want to flag for the conversation something you mentioned in person earlier, about deciding which metrics here we want to track and if there's anything we can do for SEO. My product owner was just noticing that Google searching "Connected Accounts VA.org" brings up MyHealtheVet.

**Comment by Samara-Strauss** [comment]
@mshea0606 @careyotto the ticket management on this ticket is confusing. Is this an epic? I thought this was *just* to track the design updates. If this is **not** an epic, the following should be removed from the task list in the first comment:

- Lighthouse team to provide FAQ/Learn More URL
- Lighthouse team to provide up to 5 minutes of user research questions
- Analytics / SEO

If this **is** an epic, you can leave those there. **BUT** then we need to separate the discussion around the design of Connected Accounts into a separate ticket.

Whether this is an epic or not, all of these need separate tickets:

- Lighthouse team to provide FAQ/Learn More URL
- Lighthouse team to provide up to 5 minutes of user research questions
- Analytics / SEO

So, to summarize:

- Is this an epic? If yes, please create a different ticket where we can focus on design feedback. If it's not, please take out any of the non-design related items listed above from this task list so we are not managing too many things in one thread.
- Regardless of whether this is an epic or not, the FAQ page, research questions, and analytics/SEO needs to their own tickets.

Let me know if you have any questions about that.

**Comment by Samara-Strauss** [comment]
@careyotto thanks for updating the design! My latest feedback:

- The empty state feels weird to me to have the heading, description, and then nothing but blank space. Do you or @tressaellen have ideas here? 

And questions:

- Can you clarify what each "Learn more" link you have links to? There are a few and I just want to make sure I understand where they are supposed to go.

**Comment by tressaellen** [comment]
@Samara-Strauss and @careyotto working on this now!

**Comment by mshea0606** [comment]
@Samara-Strauss @tressaellen @mshea0606 @careyotto  Next steps:
 - [x] I removed the non-design related tasks you mentioned from the ticket above.
-  [x] create EPIC for all Connected Accounts Updates to keep everything in one place
        Connected Accounts Updates - EPIC #6814
- [ ] Carey to create ticket for "Lighthouse team to provide FAQ/Learn More URL"
- [ ] Carey to create ticket for "Lighthouse team to provide up to 5 minutes of user research questions"
- [ ] Carey to create ticket for "Analytics / SEO"

**Comment by Samara-Strauss** [comment]
Thanks @mshea0606! I appreciate that. Looks like things should be cleaner, more organized, and easier to follow this way.

**Comment by tressaellen** [comment]
@careyotto and @Samara-Strauss here's my updates and a few thoughts.

1. I do not think that the disconnect should be buried - it's the only way that a user can 'edit' their account and it doesn't makes sense to me that it's hard to get to - thoughts?
2. The messaging for successfully disconnecting an account feels excessive and slightly redundant to have the link to FAQs again there - I've included a mock with a disappearing banner and I wanted to get your thoughts on that as well.
3. I've used the dropdown for extra content inside the boxes for 'what can [app] see?', is this an appropriate use according to the design system.  It made sense to me- but I wanted to double check.

<img width="381" alt="Screen Shot 2020-03-12 at 10 15 58 AM" src="https://user-images.githubusercontent.com/58752772/76554650-14a67700-644b-11ea-97f9-3b8c95807c9f.png">
<img width="390" alt="Screen Shot 2020-03-12 at 10 16 04 AM" src="https://user-images.githubusercontent.com/58752772/76554655-15d7a400-644b-11ea-87fb-1265a0843f3b.png">
<img width="411" alt="Screen Shot 2020-03-12 at 10 16 19 AM" src="https://user-images.githubusercontent.com/58752772/76554657-16703a80-644b-11ea-8646-976bd588a86d.png">
<img width="405" alt="Screen Shot 2020-03-12 at 10 16 29 AM" src="https://user-images.githubusercontent.com/58752772/76554659-17a16780-644b-11ea-9853-2b77191f3634.png">
<img width="391" alt="Screen Shot 2020-03-12 at 10 16 34 AM" src="https://user-images.githubusercontent.com/58752772/76554662-1839fe00-644b-11ea-8b31-e54282daab0b.png">

**Comment by careyotto** [comment]
Thanks @tressaellen! I agree with your concerns about burying the disconnect feature and love that you are exploring this.

**Questions/feedback:** 

1. Would a button work here for "Disconnect" rather than a link? I feel like it's currently losing prominence to the "What can Apple HealthKit view?" link

2. There is currently a typo in the "What can..." sentence. 'can' is repeated twice.

3. Re: language: "What can [app/site] view?" -- "View" may not be the right word to use. Some of the scopes described here will be things like "offline access," which means the app can continue to get new access tokens without the user being interactively engaged. FYSA: In future iterations, we may want to add links to an app's privacy policy and terms of service here, so if the Profile 2.0 design can scale to fit that, rad.

4. I appreciate the simplicity of the second confirmation design you shared, with the light green box. I'm wondering if we should also be indicating that in some cases, like with Apple health, any data that's already been retrieved could remain with the app after disconnecting (depending on how the app handles it). I don't want the msg to be lost in FAQs, but maybe it's totally fine just there? What do you and Peggy think? If we decide to not mention account deletion here, I'll def want to ask users about their expectations during user testing.

**Comment by tressaellen** [comment]
@careyotto thank you for this feedback! 

1. I tried buttons for 'disconnect' -i'll attach it here. It feels heavy to me but I think it might still make sense to use from a hierarchy perspective.
2.  Let me play around with this and see if I can come up with something a little more scalable.
![Account 1 Copy 5](https://user-images.githubusercontent.com/58752772/76560956-30177f00-6457-11ea-92f3-5193fc7f064c.jpg)
![Account 1 Copy 6](https://user-images.githubusercontent.com/58752772/76560978-39085080-6457-11ea-8431-fa806d31942e.jpg)


4. I think it's probably better to err on the side of caution and assume that the user won't know that there's the possibility of data remaining with the app - I'd probably say we just leave the full messaging if you think it's important.

**Comment by tressaellen** [comment]
@Samara-Strauss and @careyotto  the more I'm working on connected accounts I really think it's best if we break it out into it's own section within profile - especially with the future features that Carey would like to include.  I just don't want to start adding things in the future and have the whole account page be overcrowded/confusing when we can just address that issue now.  Do you guys have strong feelings either way?

**Comment by careyotto** [comment]
@tressaellen Let's hop on a call tomorrow to discuss what you're thinking. My calendar is up to date.

**Comment by Samara-Strauss** [comment]
1. Let's break connected accounts out into its own section. Can we check in with Peggy to see if now we want to call "Account" --> "Account Security" again? I think that might be more clear given what will be in there.
2. Re: buttons or links for disconnect — I think the buttons look fine if they're right aligned (option 2).
3. I also like the simplicity of the second "saved" state — is this what we're using elsewhere in the profile? Is this a new design system pattern?

Sorry if I missed something since there are a few comments above :) Let me know if there's another question that needs answering!

**Comment by tressaellen** [comment]
@Samara-Strauss thank you! The saved state is in the design system and will also be used in the profile... we're still trying to figure out messaging around it.

**Comment by tressaellen** [comment]
![Connected Accounts 2](https://user-images.githubusercontent.com/58752772/77565895-bda49700-6e79-11ea-943e-d0cbb0a7edcb.jpg)
![Connected Accounts 3](https://user-images.githubusercontent.com/58752772/77565909-c1381e00-6e79-11ea-8d0c-8a8dfe57e9d3.jpg)

**Comment by Samara-Strauss** [comment]
Is the design done? If so, we can probably close this since I know copy is being worked on in another ticket.
